<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Statistical Comparisons with comp_means • Pmisc</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Statistical Comparisons with comp_means">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">Pmisc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/comp_means.html">Statistical Comparisons with comp_means</a></li>
    <li><a class="dropdown-item" href="../articles/survival_analysis.html">Survival Analysis with Pmisc</a></li>
    <li><a class="dropdown-item" href="../articles/upset_plots.html">Set Visualization with Upset Plots</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Statistical Comparisons with comp_means</h1>
                        <h4 data-toc-skip class="author">Alyssa
Pybus</h4>
            
            <h4 data-toc-skip class="date">2026-01-12</h4>
      

      <div class="d-none name"><code>comp_means.Rmd</code></div>
    </div>

    
    
<p><strong><a href="https://github.com/afpybus/Pmisc" class="external-link">GitHub
Repository</a></strong></p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <code><a href="../reference/comp_means.html">comp_means()</a></code> function in Pmisc provides a
streamlined way to perform <strong>pairwise statistical
comparisons</strong> across multiple features <strong>between two
groups</strong>. It automatically handles:</p>
<ul>
<li>Statistical tests for two-group comparisons (t-test, Wilcoxon)</li>
<li>P-value adjustment for multiple testing</li>
<li>Calculation of fold changes and mean differences</li>
<li>Sample size validation</li>
</ul>
<p><strong>Important:</strong> This function is designed for comparing
exactly <strong>two groups</strong>. For multi-group comparisons, filter
your data to two groups at a time.</p>
<p>It wraps the powerful <strong><code>ggpubr</code></strong> package
functions to provide a more opinionated, high-throughput workflow.</p>
<p>This vignette demonstrates the function using the classic
<code>iris</code> dataset, comparing two species at a time.</p>
</div>
<div class="section level2">
<h2 id="load-package">Load Package<a class="anchor" aria-label="anchor" href="#load-package"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">Pmisc</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.5.2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.5.2</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: package 'tibble' was built under R version 4.5.2</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="simulating-gene-expression-data">Simulating Gene Expression Data<a class="anchor" aria-label="anchor" href="#simulating-gene-expression-data"></a>
</h2>
<p>To demonstrate the function’s capability with high-dimensional data,
let’s simulate a gene expression dataset with 50 genes across two
experimental groups: “Control” and “Treatment”.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span> <span class="co"># For reproducibility</span></span>
<span></span>
<span><span class="co"># 1. Create sample metadata</span></span>
<span><span class="va">n_samples</span> <span class="op">&lt;-</span> <span class="fl">20</span> <span class="co"># 10 per group</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    sample_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"S"</span>, <span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span><span class="op">)</span>,</span>
<span>    group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Control"</span>, <span class="st">"Treatment"</span><span class="op">)</span>, each <span class="op">=</span> <span class="va">n_samples</span> <span class="op">/</span> <span class="fl">2</span><span class="op">)</span>,</span>
<span>    stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Simulate gene expression matrix (50 genes)</span></span>
<span><span class="co"># Most genes will be unchanged, some will be differentially expressed</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Gene_"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">)</span></span>
<span><span class="va">expression_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="va">n_samples</span> <span class="op">*</span> <span class="fl">50</span>, mean <span class="op">=</span> <span class="fl">10</span>, sd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="va">n_samples</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">expression_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">genes</span></span>
<span></span>
<span><span class="co"># Convert to data frame and add group info</span></span>
<span><span class="va">df_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">metadata</span>, <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">expression_data</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 3. Introduce signal (Differential Expression)</span></span>
<span><span class="co"># Upregulate 5 genes in Treatment</span></span>
<span><span class="va">up_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Gene_1"</span>, <span class="st">"Gene_2"</span>, <span class="st">"Gene_3"</span>, <span class="st">"Gene_4"</span>, <span class="st">"Gene_5"</span><span class="op">)</span></span>
<span><span class="va">df_genes</span><span class="op">[</span><span class="va">df_genes</span><span class="op">$</span><span class="va">group</span> <span class="op">==</span> <span class="st">"Treatment"</span>, <span class="va">up_genes</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">df_genes</span><span class="op">[</span><span class="va">df_genes</span><span class="op">$</span><span class="va">group</span> <span class="op">==</span> <span class="st">"Treatment"</span>, <span class="va">up_genes</span><span class="op">]</span> <span class="op">+</span> <span class="fl">3</span></span>
<span></span>
<span><span class="co"># Downregulate 5 genes in Treatment</span></span>
<span><span class="va">down_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Gene_6"</span>, <span class="st">"Gene_7"</span>, <span class="st">"Gene_8"</span>, <span class="st">"Gene_9"</span>, <span class="st">"Gene_10"</span><span class="op">)</span></span>
<span><span class="va">df_genes</span><span class="op">[</span><span class="va">df_genes</span><span class="op">$</span><span class="va">group</span> <span class="op">==</span> <span class="st">"Treatment"</span>, <span class="va">down_genes</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">df_genes</span><span class="op">[</span><span class="va">df_genes</span><span class="op">$</span><span class="va">group</span> <span class="op">==</span> <span class="st">"Treatment"</span>, <span class="va">down_genes</span><span class="op">]</span> <span class="op">-</span> <span class="fl">3</span></span>
<span></span>
<span><span class="co"># View structure</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">df_genes</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sample_id"</span>, <span class="st">"group"</span>, <span class="st">"Gene_1"</span>, <span class="st">"Gene_6"</span>, <span class="st">"Gene_50"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;   sample_id   group    Gene_1    Gene_6   Gene_50</span></span>
<span><span class="co">#&gt; 1        S1 Control  8.879049  8.579187 14.610124</span></span>
<span><span class="co">#&gt; 2        S2 Control  9.539645 10.513767  7.750793</span></span>
<span><span class="co">#&gt; 3        S3 Control 13.117417  9.506616  9.389061</span></span>
<span><span class="co">#&gt; 4        S4 Control 10.141017  9.304915  8.966481</span></span>
<span><span class="co">#&gt; 5        S5 Control 10.258575  8.096763 13.024791</span></span>
<span><span class="co">#&gt; 6        S6 Control 13.430130  9.909945  8.461030</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="running-the-comparison">Running the Comparison<a class="anchor" aria-label="anchor" href="#running-the-comparison"></a>
</h2>
<p>Now we run <code><a href="../reference/comp_means.html">comp_means()</a></code> on all 50 genes
simultaneously.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define all gene columns as features</span></span>
<span><span class="va">features</span> <span class="op">&lt;-</span> <span class="va">genes</span></span>
<span></span>
<span><span class="co"># Run comparison</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/comp_means.html">comp_means</a></span><span class="op">(</span></span>
<span>    df <span class="op">=</span> <span class="va">df_genes</span>,</span>
<span>    feature_column_names <span class="op">=</span> <span class="va">features</span>,</span>
<span>    group_column_name <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>    compare_means_method <span class="op">=</span> <span class="st">"t.test"</span>,</span>
<span>    p.adjust_method <span class="op">=</span> <span class="st">"fdr"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View top significant results</span></span>
<span><span class="va">results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">p.adj</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">feature</span>, <span class="va">group1</span>, <span class="va">group2</span>, <span class="va">mean_dif</span>, <span class="va">p.adj</span>, <span class="va">p.adj.signif</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>digits <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">feature</th>
<th align="left">group1</th>
<th align="left">group2</th>
<th align="right">mean_dif</th>
<th align="right">p.adj</th>
<th align="left">p.adj.signif</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Gene_2</td>
<td align="left">Control</td>
<td align="left">Treatment</td>
<td align="right">-4.4932</td>
<td align="right">0.0005</td>
<td align="left">***</td>
</tr>
<tr class="even">
<td align="left">Gene_6</td>
<td align="left">Control</td>
<td align="left">Treatment</td>
<td align="right">2.8541</td>
<td align="right">0.0080</td>
<td align="left">**</td>
</tr>
<tr class="odd">
<td align="left">Gene_10</td>
<td align="left">Control</td>
<td align="left">Treatment</td>
<td align="right">3.5345</td>
<td align="right">0.0087</td>
<td align="left">**</td>
</tr>
<tr class="even">
<td align="left">Gene_3</td>
<td align="left">Control</td>
<td align="left">Treatment</td>
<td align="right">-3.4608</td>
<td align="right">0.0099</td>
<td align="left">**</td>
</tr>
<tr class="odd">
<td align="left">Gene_5</td>
<td align="left">Control</td>
<td align="left">Treatment</td>
<td align="right">-3.2480</td>
<td align="right">0.0099</td>
<td align="left">**</td>
</tr>
<tr class="even">
<td align="left">Gene_1</td>
<td align="left">Control</td>
<td align="left">Treatment</td>
<td align="right">-3.2680</td>
<td align="right">0.0149</td>
<td align="left">*</td>
</tr>
<tr class="odd">
<td align="left">Gene_9</td>
<td align="left">Control</td>
<td align="left">Treatment</td>
<td align="right">3.4942</td>
<td align="right">0.0186</td>
<td align="left">*</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="volcano-plot-visualization">Volcano Plot Visualization<a class="anchor" aria-label="anchor" href="#volcano-plot-visualization"></a>
</h2>
<p>With 50 variables, the volcano plot becomes a powerful tool to
visualize the overall landscape of differential expression.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define colors: Control (blue), Treatment (red), ns (gray)</span></span>
<span><span class="co"># Note: "Control" is the reference, so "Treatment" is the comparison group</span></span>
<span><span class="va">sig_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Control"</span>, <span class="st">"Treatment"</span>, <span class="st">"ns"</span><span class="op">)</span>,</span>
<span>    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"dodgerblue"</span>, <span class="st">"firebrick"</span>, <span class="st">"grey80"</span><span class="op">)</span>,</span>
<span>    stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create volcano plot</span></span>
<span><span class="fu"><a href="../reference/volcano_cm.html">volcano_cm</a></span><span class="op">(</span></span>
<span>    comp_means_output <span class="op">=</span> <span class="va">results</span>,</span>
<span>    x <span class="op">=</span> <span class="st">"mean_dif"</span>,</span>
<span>    max_overlaps <span class="op">=</span> <span class="fl">20</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="../reference/scm.html">scm</a></span><span class="op">(</span><span class="va">sig_colors</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Differential Expression: Treatment vs Control"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in geom_text(aes(x = max(x_var) - 0.07 * sum(range(abs(x_var))), : All aesthetics have length 1, but the data has 50 rows.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please consider using `annotate()` or provide this layer with data containing</span></span>
<span><span class="co">#&gt;   a single row.</span></span></code></pre></div>
<p><img src="figures/comp_means-volcano_plot-1.png" class="r-plt" alt="" width="672"></p>
<p>The plot clearly shows our simulated signal: - <strong>Red points
(Right)</strong>: Genes upregulated in Treatment (Gene_1 to Gene_5) -
<strong>Blue points (Left)</strong>: Genes higher in Control (meaning
downregulated in Treatment) (Gene_6 to Gene_10) - <strong>Gray points
(Bottom/Middle)</strong>: Unchanged background genes</p>
</div>
<div class="section level2">
<h2 id="visualizing-top-hits">Visualizing Top Hits<a class="anchor" aria-label="anchor" href="#visualizing-top-hits"></a>
</h2>
<p>We can verify our top hits using boxplots.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize top upregulated gene</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DEF_boxplot.html">DEF_boxplot</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">df_genes</span>,</span>
<span>    DE <span class="op">=</span> <span class="va">results</span>,</span>
<span>    feature <span class="op">=</span> <span class="st">"Gene_1"</span>, <span class="co"># Known upregulated</span></span>
<span>    grouping <span class="op">=</span> <span class="st">"group"</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Gene_1 (Upregulated)"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize top downregulated gene</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DEF_boxplot.html">DEF_boxplot</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">df_genes</span>,</span>
<span>    DE <span class="op">=</span> <span class="va">results</span>,</span>
<span>    feature <span class="op">=</span> <span class="st">"Gene_6"</span>, <span class="co"># Known downregulated</span></span>
<span>    grouping <span class="op">=</span> <span class="st">"group"</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Gene_6 (Downregulated)"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Arrange side by side</span></span>
<span><span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="figures/comp_means-boxplot-1.png" class="r-plt" alt="" width="672"></p>
</div>
<div class="section level2">
<h2 id="non-parametric-tests-wilcoxon">Non-Parametric Tests (Wilcoxon)<a class="anchor" aria-label="anchor" href="#non-parametric-tests-wilcoxon"></a>
</h2>
<p>We can also perform the same broad comparison using the Wilcoxon
test.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run Wilcoxon test on all genes</span></span>
<span><span class="va">results_wilcox</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/comp_means.html">comp_means</a></span><span class="op">(</span></span>
<span>    df <span class="op">=</span> <span class="va">df_genes</span>,</span>
<span>    feature_column_names <span class="op">=</span> <span class="va">features</span>,</span>
<span>    group_column_name <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>    compare_means_method <span class="op">=</span> <span class="st">"wilcox.test"</span>,</span>
<span>    p.adjust_method <span class="op">=</span> <span class="st">"fdr"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare p-values for a specific gene (e.g., Gene_1)</span></span>
<span><span class="va">comparison</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    feature <span class="op">=</span> <span class="st">"Gene_1"</span>,</span>
<span>    t_test_p <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/format.pval.html" class="external-link">format.pval</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">p.adj</span><span class="op">[</span><span class="va">results</span><span class="op">$</span><span class="va">feature</span> <span class="op">==</span> <span class="st">"Gene_1"</span><span class="op">]</span>, digits <span class="op">=</span> <span class="fl">3</span>, eps <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span>,</span>
<span>    wilcox_p <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/format.pval.html" class="external-link">format.pval</a></span><span class="op">(</span><span class="va">results_wilcox</span><span class="op">$</span><span class="va">p.adj</span><span class="op">[</span><span class="va">results_wilcox</span><span class="op">$</span><span class="va">feature</span> <span class="op">==</span> <span class="st">"Gene_1"</span><span class="op">]</span>, digits <span class="op">=</span> <span class="fl">3</span>, eps <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">comparison</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">feature</th>
<th align="left">t_test_p</th>
<th align="left">wilcox_p</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">Gene_1</td>
<td align="left">0.0149</td>
<td align="left">0.0328</td>
</tr></tbody>
</table>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>The <code><a href="../reference/comp_means.html">comp_means()</a></code> function provides:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Flexible testing</strong>: Support for t-test, Wilcoxon, and
ANOVA</li>
<li>
<strong>Multiple testing correction</strong>: FDR, Bonferroni, Holm,
and others</li>
<li>
<strong>Rich output</strong>: Means, fold changes, p-values, and
significance labels</li>
<li>
<strong>Integration</strong>: Works seamlessly with volcano plots
and boxplots</li>
<li>
<strong>Sample size checking</strong>: Automatically removes
features with insufficient data</li>
</ol>
<p>This makes it ideal for exploratory analysis, differential expression
studies, and comparing measurements across experimental groups.</p>
</div>
<div class="section level2">
<h2 id="related-functions">Related Functions<a class="anchor" aria-label="anchor" href="#related-functions"></a>
</h2>
<ul>
<li>
<code><a href="../reference/p.readjust.html">p.readjust()</a></code>: Re-adjust p-values from comp_means
output</li>
<li>
<code><a href="../reference/volcano_cm.html">volcano_cm()</a></code>: Create volcano plots</li>
<li>
<code><a href="../reference/volcano_cm_labelall.html">volcano_cm_labelall()</a></code>: Volcano plot labeling all
points</li>
<li>
<code><a href="../reference/DEF_boxplot.html">DEF_boxplot()</a></code>: Boxplot with p-value annotations</li>
<li>
<code><a href="../reference/DEF_boxplot_sig.html">DEF_boxplot_sig()</a></code>: Boxplot with significance stars</li>
</ul>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alyssa Pybus.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
